### ISSUE TITLE
Escaping Issue on Windows

### ISSUE THREAD

**Issue by nathanhammond** [IndigoUnited/node-cross-spawn-async#17]
Hey @satazor!

[One of your recent changes](https://github.com/IndigoUnited/node-cross-spawn-async/commit/26032206d9fc87105f5467a515ac6e4fed7cb1ba) resulted in failing tests inside of ember-cli. We don't have a reduced failure case as it may be an issue in pair with `phantomjs` which may expect different input.

At 2.2.1 (not working):
```javascript
{ command: 'C:\\Windows\\system32\\cmd.exe',                                                                   
  args:                                                                                                        
   [ '/s',                                                                                                     
     '/c',                                                                                                     
     '"phantomjs C:\\Users\\IEUser\\Code\\some-cool-app\\node_modules\\ember-cli\\node_modules\\testem\\assets\\phantom.js http://localhost:7357/440/tests/index.html?hidepassed=^&filter=jshint"' ],
  options: { windowsVerbatimArguments: true },                                                                 
  file: 'C:\\Users\\IEUser\\AppData\\Roaming\\npm\\phantomjs.CMD',                                             
  original: 'phantomjs' }                                                                                      
```

At 2.1.9 (working):
```javascript
{ command: 'C:\\Windows\\system32\\cmd.exe',
  args:
   [ '/s',
     '/c',
     '"phantomjs "C:\\Users\\IEUser\\Code\\some-cool-app\\node_modules\\ember-cli\\node_modules\\testem\\assets\\phantom.js" "http://localhost:7357/4137/tests/index.html?hidepassed=&filter=jshint""' ],
  options: { windowsVerbatimArguments: true },
  file: 'C:\\Users\\IEUser\\AppData\\Roaming\\npm\\phantomjs.CMD',
  original: 'phantomjs' }
```

Note the tiniest difference in the `^&` (you'll have to scroll the code fields). We're working around this by pinning a dependency. Sorry not to have a reduced test case.

/cc @Turbo87

**Comment by Turbo87** [comment]
for future reference we're tracking this in https://github.com/ember-cli/ember-cli/issues/5781

**Comment by satazor** [comment]
I'm sorry for any inconvenience, I will look at this when I arrive home.

**Comment by nathanhammond** [comment]
We're not bothered! :) Our goal is to help make the entire ecosystem solid.

**Comment by satazor** [comment]
@nathanhammond there's nothing wrong with the ^, it's an escape meta char and its being interpreted correctly. By analyzing the both commands (before and after), there's another difference: 440  vs 4137 in the URL. Are these intended?

I would like to get some instructions in how to replicate the issue in my laptop. i've run the ember-cli test suite on a windows VM with an .only on the failing test but it's hanging, see: https://github.com/ember-cli/ember-cli/issues/5781#issuecomment-209126135

**Comment by nathanhammond** [comment]
@satazor The number shouldn't make a difference. I just noticed that there is also some sort of nested double quoting here:

```
'"phantomjs C:\\Users\\IEUser\\Code\\some-cool-app\\node_modules\\ember-cli\\node_modules\\testem\\assets\\phantom.js http://localhost:7357/1111/tests/index.html?hidepassed=^&filter=jshint"' ],
'"phantomjs "C:\\Users\\IEUser\\Code\\some-cool-app\\node_modules\\ember-cli\\node_modules\\testem\\assets\\phantom.js" "http://localhost:7357/1111/tests/index.html?hidepassed=&filter=jshint""' ],
```

I'm completely unfamiliar with escaping for Windows CLI (and on a Mac), so I'm hardly useful here. I only know that these two commands result in different outcomes for phantomjs. It's also possible that this is a phantomjs bug in its Windows option parsing assuming that the escaping is correct.

**Comment by Turbo87** [comment]
@satazor you can reproduce if you clone https://github.com/Turbo87/some-cool-app, then check out the second commit and run `node_modules/.bin/ember test --filter=jshint`. this causes the url to include `?hidepassed=&filter=jshint` but the `filter` parameter is actually not recognized due to the `&` being misinterpreted.

**Comment by satazor** [comment]
The commit I've made was reverted and released in `2.2.2`. Again sorry guys!

**Comment by Turbo87** [comment]
@satazor awesome, thanks for the fast fix!
